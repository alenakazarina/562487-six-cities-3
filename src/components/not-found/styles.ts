export const WrapperStyles: React.CSSProperties = {
  width: `420px`,
  marginTop: `16.7vh`,
  marginRight: `auto`,
  marginLeft: `auto`,
  paddingTop: `94px`,
  textAlign: `center`,
  backgroundImage: `url(../img/ico-saved.svg)`,
  backgroundSize: `60px 73px`,
  backgroundPosition: `center top`,
  backgroundRepeat: `no-repeat`
};

export const StatusStyles: React.CSSProperties = {
  display: `block`,
  marginBottom: `5px`,
  fontSize: `32px`,
  lineHeight: `1.1875`
};

export const DescriptionStyles: React.CSSProperties = {
  marginTop: `0`,
  padding: `0 30px`,
  fontSize: `16px`,
  lineHeight: `1.5`
};

export const CityStyles: React.CSSProperties = {
  display: `flex`,
  justifyContent: `center`,
  marginTop: `20px`
};
