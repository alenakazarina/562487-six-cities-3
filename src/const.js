export const BASE_URL = `https://htmlacademy-react-3.appspot.com/six-cities`;

export const SortType = {
  POPULAR: `Popular`,
  PRICE_TO_HIGH: `Price: low to high`,
  PRICE_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`
};

export const OfferType = {
  house: `House`,
  room: `Private room`,
  apartment: `Apartment`,
  hotel: `Hotel`
};

export const AppRoute = {
  ROOT: `/`,
  OFFER: `/offer/:id`,
  FAVORITES: `/favorites`,
  LOGIN: `/login`
};

export const RATINGS = [
  {
    value: 5,
    title: `perfect`
  },
  {
    value: 4,
    title: `good`
  },
  {
    value: 3,
    title: `not bad`
  },
  {
    value: 2,
    title: `badly`
  },
  {
    value: 1,
    title: `terribly`
  }
];

export const MONTHS = [
  `January`, `February`, `March`, `April`,
  `May`, `June`, `July`, `August`,
  `September`, `October`, `November`, `December`
];

export const AuthStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const InitialState = {
  USER: {
    authStatus: `NO_AUTH`,
    user: {
      id: -1,
      email: ``,
      name: ``,
      isPro: false,
      avatarUrl: ``
    }
  },
  FAVORITES: {
    favorites: []
  },
  OFFER: {
    activeOffer: null,
    nearOffers: [],
    comments: []
  },
  OFFERS: {
    offers: [],
    activeCity: ``
  },
  ERRORS: {
    errorStatus: null
  }
};

