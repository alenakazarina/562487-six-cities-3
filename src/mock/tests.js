export const locations = [
  {
    title: `Paris`,
    isActive: true
  },
  {
    title: `Cologne`,
    isActive: false
  },
  {
    title: `Brussels`,
    isActive: false
  },
  {
    title: `Amsterdam`,
    isActive: false
  },
  {
    title: `Hamburg`,
    isActive: false
  },
  {
    title: `Dusseldorf`,
    isActive: false
  }
];

export const offers = [
  {
    id: `01`,
    title: `Beautiful & luxurious apartment at great location`
  },
  {
    id: `02`,
    title: `Wood and stone place`
  },
  {
    id: `03`,
    title: `Canal View Prinsengracht`
  },
  {
    id: `04`,
    title: `Nice, cozy, warm big bed apartment`
  },
  {
    id: `05`,
    title: `Cozy rooms in the city center`
  }
];

export const title = `Canal View Prinsengracht`;

export const onTitleClick = () => {};
