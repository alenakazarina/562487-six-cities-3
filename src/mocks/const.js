export const IMAGE_PATH = `./img/`;

const LOCATIONS_ZOOM = 13;

export const MONTHS = [
  `January`, `February`, `March`, `April`, `May`, `June`, `July`,
  `August`, `September`, `October`, `November`, `December`
];

const OfferTypes = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`
};

export const OffersData = {
  maxRating: 5,
  types: Object.values(OfferTypes),
  titles: [
    `Canal View room`,
    `Nice, cozy, warm big bed apartment`,
    `Wood and stone place`,
    `Beautiful & luxurious apartment at great location`,
    `Cozy rooms in the city center`
  ],
  images: [
    `apartment-01.jpg`,
    `apartment-02.jpg`,
    `apartment-03.jpg`,
    `apartment-04.jpg`,
    `apartment-05.jpg`,
  ],
  amenities: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Cabel TV`, `Fridge`],
  description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
};

export const UserData = {
  names: [`Douglas`, `James`, `Matthew`, `John`, `Leo`, `Robert`, `Dirk`],
  avatarUrls: [
    `user-1.jpg`,
    `user-2.jpg`,
    `user-3.jpg`,
    `user-4.jpg`,
    `user-5.jpg`,
  ],
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
};

export const LOCATIONS = [
  {
    name: `Paris`,
    location: {
      latitude: 48.856663,
      longitude: 2.351556,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Cologne`,
    location: {
      latitude: 50.930779,
      longitude: 6.938399,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Brussels`,
    location: {
      latitude: 50.851309,
      longitude: 4.351718,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Amsterdam`,
    location: {
      latitude: 52.373057,
      longitude: 4.892557,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Hamburg`,
    location: {
      latitude: 53.552645,
      longitude: 9.966287,
      zoom: LOCATIONS_ZOOM
    }
  }, {
    name: `Dusseldorf`,
    location: {
      latitude: 51.230569,
      longitude: 6.787428,
      zoom: LOCATIONS_ZOOM
    }
  }];
