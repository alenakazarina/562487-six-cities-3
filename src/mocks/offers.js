import {OffersData} from './const';
import {
  generateId,
  getImage,
  getRandomBoolean,
  getRandomNumber,
  getRandomFromArray,
} from './utils';

const generateOffer = () => {
  return {
    id: generateId(),
    previewImage: getImage(OffersData.images),
    title: getRandomFromArray(OffersData.titles),
    isFavorite: getRandomBoolean(),
    isPremium: getRandomBoolean(),
    rating: getRandomNumber({min: 0, max: OffersData.maxRating}),
    type: getRandomFromArray(OffersData.types),
    price: getRandomNumber({min: 0, max: 1000})
  };
};

export const generateOffers = (count = 3) => {
  return new Array(count).fill(``).map(generateOffer);
};

export const offers = [
  {
    city: `Paris`,
    offers: generateOffers(4)
  },
  {
    city: `Cologne`,
    offers: generateOffers(5)
  },
  {
    city: `Brussels`,
    offers: generateOffers(4)
  },
  {
    city: `Amsterdam`,
    offers: generateOffers(5)
  },
  {
    city: `Hamburg`,
    offers: generateOffers(4)
  },
  {
    city: `Dusseldorf`,
    offers: generateOffers(5)
  }
];
