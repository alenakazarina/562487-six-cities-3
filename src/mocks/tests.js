import configureStore from 'redux-mock-store';
import {AuthStatus} from '../reducers/user/user';
import {cityOffers, reviews, appUsers} from './const';

const defaultUser = {
  id: -1,
  email: ``,
  name: ``,
  isPro: false,
  avatarUrl: ``
};

const mockStore = configureStore([]);

export const storeWithAuth = mockStore({
  USER: {
    authStatus: AuthStatus.AUTH,
    user: appUsers[0]
  },
  OFFERS: {
    offers: cityOffers,
    activeCity: `Paris`
  },
  OFFER: {
    pageOffer: cityOffers[1],
    activeOffer: cityOffers[2],
    nearOffers: cityOffers.slice(0, 2),
    comments: reviews
  },
  FAVORITES: {
    favorites: cityOffers
  }
});

export const storeWithNoAuth = mockStore({
  USER: {
    authStatus: AuthStatus.NO_AUTH,
    user: defaultUser
  },
  OFFERS: {
    offers: cityOffers,
    activeCity: `Paris`
  },
  OFFER: {
    pageOffer: null,
    activeOffer: null,
    nearOffers: [],
    comments: []
  },
  FAVORITES: {
    favorites: []
  }
});

export const storeWithNoOffers = mockStore({
  USER: {
    authStatus: AuthStatus.NO_AUTH,
    user: defaultUser
  },
  OFFERS: {
    offers: [],
    activeCity: `Paris`
  },
  OFFER: {
    pageOffer: null,
    activeOffer: null,
    nearOffers: [],
    comments: []
  },
  FAVORITES: {
    favorites: []
  }
});
