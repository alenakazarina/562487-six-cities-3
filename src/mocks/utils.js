import nanoid from 'nanoid';
import {IMAGE_PATH, OffersData} from './const';

const getRandomNumber = ({min = 0, max = 1}) => {
  return min + Math.floor(max * Math.random());
};

const getRandomFromObject = (obj = {}) => {
  const randomKey = getRandomFromArray(Object.keys(obj));
  return obj[randomKey];
};

const getRandomFromArray = (arr = []) => {
  const randomIndex = getRandomNumber({min: 0, max: arr.length});
  return arr[randomIndex];
};

const getRandomArray = ({items = [], minLength = 0, maxLength = 1}) => {
  const length = getRandomNumber({min: minLength, max: maxLength + 1});
  return new Array(length).fill(``).map(() => {
    const index = getRandomNumber(0, items.length);
    return items[index];
  });
};

const getRandomBoolean = () => {
  return Math.random() > 0.5;
};

const generateId = () => {
  return nanoid();
};

const getImage = (items = []) => {
  const randomIndex = getRandomNumber({min: 0, max: items.length});
  return IMAGE_PATH.concat(items[randomIndex]);
};

const generateImages = () => {
  return OffersData.images.map((image) => IMAGE_PATH.concat(image));
};

const formatMonthYear = (date = new Date()) => {
  return `${date.getMonth() + 1} ${date.getFullYear()}`;
};

export {
  getRandomNumber,
  getRandomArray,
  getRandomBoolean,
  generateId,
  getImage,
  generateImages,
  getRandomFromArray,
  getRandomFromObject,
  formatMonthYear
};
