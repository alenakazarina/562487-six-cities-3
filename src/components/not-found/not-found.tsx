import * as React from 'react';
import {connect} from 'react-redux';
import {AppUser} from '../../types';
import {getActiveCity} from '../../reducers/offers/selectors';
import Header from '../../components/header/header';
import LocationsListItem from '../locations-list-item/locations-list-item';
import {WrapperStyles, StatusStyles, DescriptionStyles, CityStyles} from './styles';

interface Props {
  isAuth: boolean;
  user: AppUser;
  activeCity: string;
}

const NotFound: React.FC<Props> = (props: Props) => {
  const {
    isAuth,
    user,
    activeCity
  } = props;
  return (
    <div className="page page--gray page--not-found">
      <Header
        isAuth={isAuth}
        user={user}
      />
      <main className="page__not-found">
        <div className="not-found__status-wrapper" style={WrapperStyles}>
          <b className="not-found__status" style={StatusStyles}>Page Not Found</b>
          <p className="not-found__status-description"
            style={DescriptionStyles}
          >It happens! Letâ€™s get you back on track.</p>
          <div style={CityStyles}>
            <LocationsListItem
              nodeType="div"
              city={activeCity}
              isActive
            />
          </div>
        </div>
      </main>
    </div>
  );
};

const mapStateToProps = (state) => ({
  activeCity: getActiveCity(state)
});

export {NotFound};
export default connect(mapStateToProps)(NotFound);
