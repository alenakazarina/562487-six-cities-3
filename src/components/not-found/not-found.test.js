import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {BrowserRouter} from 'react-router-dom';
import {storeWithNoAuth} from '../../mocks/tests';
import {defaultUser} from '../../mocks/const';
import {NotFound} from './not-found';

const city = `Paris`;

describe(`NotFound`, () => {
  it(`should render NotFound Page correctly`, () => {
    const tree = renderer.create(
        <Provider store={storeWithNoAuth}>
          <BrowserRouter>
            <NotFound
              isAuth={false}
              user={defaultUser}
              activeCity={city}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
