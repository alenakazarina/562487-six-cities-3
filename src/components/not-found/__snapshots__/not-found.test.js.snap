// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should render NotFound Page correctly 1`] = `
<div
  className="page page--gray page--not-found"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="../img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__not-found"
  >
    <div
      className="not-found__status-wrapper"
      style={
        Object {
          "backgroundImage": "url(../img/ico-saved.svg)",
          "backgroundPosition": "center top",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "60px 73px",
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": "16.7vh",
          "paddingTop": "94px",
          "textAlign": "center",
          "width": "420px",
        }
      }
    >
      <b
        className="not-found__status"
        style={
          Object {
            "display": "block",
            "fontSize": "32px",
            "lineHeight": "1.1875",
            "marginBottom": "5px",
          }
        }
      >
        Page Not Found
      </b>
      <p
        className="not-found__status-description"
        style={
          Object {
            "fontSize": "16px",
            "lineHeight": "1.5",
            "marginTop": "0",
            "padding": "0 30px",
          }
        }
      >
        It happens! Letâ€™s get you back on track.
      </p>
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "center",
            "marginTop": "20px",
          }
        }
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item tabs__item--active"
            href="/"
            onClick={[Function]}
          >
            <span>
              Paris
            </span>
          </a>
        </div>
      </div>
    </div>
  </main>
</div>
`;
